feat: React docs (#5399)
PostHog/posthog.com: contents/tutorials/nextjs-analytics.md

## Adding PostHog

> At this point, you need a PostHog instance ([signup for free](https://app.posthog.com/signup)). Once created, get your PostHog API key which is found in the getting started flow or your project settings and add it to your `.env.local` file and NextJS environment variables.

Reviewer: ```suggestion
At this point, you need a PostHog instance ([signup for free](https://app.posthog.com/signup)). Once created, get your PostHog API key which is found in the getting started flow or your project settings and add it to your `.env.local` file and Next.js environment variables.
```

> ```bash

Reviewer: ```suggestion
```shell file=.env.local
```

import Script from "next/script"
import { useRouter } from "next/router"
import { useEffect } from "react"
import posthog from "posthog-js"

> import { PostHogProvider } from 'posthog-js/react'

Reviewer: ```suggestion
import { PostHogProvider } from "posthog-js/react"
```

export default function App(
  { Component, pageProps: { session, ...pageProps } }
) {
  const router = useRouter()
  useEffect(() => {
    // Track page views
    const handleRouteChange = () => posthog.capture('$pageview')
    router.events.on('routeChangeComplete', handleRouteChange)

    return () => {
        router.events.off('routeChangeComplete', handleRouteChange)
    }

> }, [])

Reviewer: ```suggestion
  }, [])
```

### Client-side rendering feature flags

> To implement this flag in our app, start with client-side rendering. 

Reviewer: much nicer 😅
